File: app/News/go/News/changePass.go
// Copyright 06-Jul-2020 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

// Change pass javascript page
package main

import (
	"github.com/dedeme/golib/cgi"
	"github.com/dedeme/golib/json"
)

func changePassProcess(ck string, mrq map[string]json.T) string {
	user := cgi.RqString(mrq, "user")
	old := cgi.RqString(mrq, "old")
	new := cgi.RqString(mrq, "new")

	return cgi.ChangePass(ck, user, old, new)
}

File: app/MMarket_old/c/src/server/settings/changepass.c
// Copyright 16-Mar-2020 ºDeme
// GNU General Public License - V3 <http://www.gnu.org/licenses/>

#include "server/settings/changepass.h"
#include "dmc/cgi.h"

char *changepass_process (AsyncActor *ac, Map *mrq) {
  CGI_GET_STR(user, mrq);
  CGI_GET_STR(old, mrq);
  CGI_GET_STR(new, mrq);

  char *rp;
  void fn () { rp = cgi_change_pass(user, old, new); }
  asyncActor_wait(ac, fn);
  return rp;
}

